### PR TITLE
chore: check not None path on datasamplespec

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Related issue

`#` followed by the number of the issue

## Please check if the PR fulfills these requirements

- [ ] If necessary, the [changelog](https://github.com/Substra/substra/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification
- For any breaking changes, companion PRs have been opened on the following repositories:
  - [ ] [substra-tests](https://github.com/Substra/substra)
  - [ ] [substrafl](https://github.com/Substra/substrafl)
  - [ ] [substra-documentation](https://github.com/Substra/substra-documentation)

### CODE REVIEW

File: substra/sdk/schemas.py [hunk 1]
```diff
@@ -146,6 +146,30 @@ class DataSampleSpec(_Spec):
     def is_many(self):
         return self.paths and len(self.paths) > 0
 
+    @pydantic.field_validator("paths")
+    @classmethod
+    def resolve_paths(cls, v: typing.Any) -> typing.Any:
```
Comment (guilhem-barthes) [inline substra/sdk/schemas.py:151]:
```suggestion
    def resolve_paths(cls, v: Optional[List[pathlib.Path]]) -> Optional[List[pathlib.Path]]:
```

File: substra/sdk/schemas.py [hunk 2]
```diff
@@ -146,6 +146,30 @@ class DataSampleSpec(_Spec):
...
+        paths = []
+        for path in v:
+            path = pathlib.Path(path)
+            paths.append(path.resolve())
+
+        return paths
+
+    @pydantic.field_validator("path")
+    @classmethod
+    def resolve_path(cls, v: typing.Any) -> typing.Any:
```
Comment (guilhem-barthes) [inline substra/sdk/schemas.py:165]:
```suggestion
    def resolve_path(cls, v: Optional[pathlib.Path]) -> Optional[pathlib.Path]:
```

File: substra/sdk/schemas.py [hunk 3]
```diff
@@ -146,6 +146,30 @@ class DataSampleSpec(_Spec):
...
+    @pydantic.field_validator("paths")
+    @classmethod
+    def resolve_paths(cls, v: typing.Any) -> typing.Any:
+        """Resolve given paths."""
+        if v is None:
+            raise ValueError("'paths' cannot be set to None.")
+
+        paths = []
+        for path in v:
+            path = pathlib.Path(path)
```
Comment (guilhem-barthes) [inline substra/sdk/schemas.py:158]:
```suggestion
```

pydantic already casts str to path when using the type `pathlib.Path`

File: substra/sdk/schemas.py [hunk 4]
```diff
@@ -146,6 +146,30 @@ class DataSampleSpec(_Spec):
...
+            paths.append(path.resolve())
+
+        return paths
```
Comment (guilhem-barthes) [inline substra/sdk/schemas.py:161]:
```suggestion
        return [p.resolve() for p in paths]
```

File: substra/sdk/schemas.py [hunk 5]
```diff
@@ -146,6 +146,30 @@ class DataSampleSpec(_Spec):
...
+
+    @pydantic.field_validator("path")
+    @classmethod
+    def resolve_path(cls, v: typing.Any) -> typing.Any:
+        """Resolve given path."""
+        if v is None:
+            raise ValueError("'path' cannot be set to None.")
+
+        path = pathlib.Path(v)
```
Comment (guilhem-barthes) [inline substra/sdk/schemas.py:170]:
```suggestion
```

**Review by guilhem-barthes — COMMENTED** [review]
Thank for your work, some comments on typing ^^

Nitpick: If path or paths are expected to be existing dirs, we could use `pydantic.DirectoryPath` (https://docs.pydantic.dev/latest/api/types/#pydantic.types.DirectoryPath)

These changes may be need to be checked on Windows (but has we never cast to `PosixPath`, I expect it won't break anything)

File: substra/sdk/schemas.py (hunk 3 above)
Comment (ThibaultFy) [reply substra/sdk/schemas.py:158]: Oh I didn't knew ! So nice...

File: substra/sdk/schemas.py (hunk 1 above)
Comment (ThibaultFy) [reply substra/sdk/schemas.py:151]: Oh great. And I removed the "optional" in that case, as, if the field is specified, this is not optional anymore to have a path. WDT ?

**Review by guilhem-barthes — APPROVED** [review]
LGTM

**Pull request merged**